pyproject.toml

```diff
@@ -48,6 +48,21 @@ dev = [
     "build>=0.10.0",
     "pytest-cov>=4.1.0",
     "pyyaml>=6.0.1",
+    "setuptools-scm[toml]>=7.1",
```

Comment:
This is already listed in the build system requirements (L2), so why is it necessary here?

Reply:
I'd also prefer not to have it here again, but it's needed to derive the current version number from the CLI when building the docs in the CI pipeline (`python -m setuptools_scm`).

I'm open to other suggestions to get it (e.g., from `build`, but that doesn't seem to support doing anything other than building the wheel/sdist, which would be overkill when trying to just get the version number).

Reply:
Fair enough, this should be the same package in any reasonable scenario.

(The only reasonable alternative would I guess be a raw `python -c "import lakefs_spec.__version__"` or something, but I can see why this is not as attractive)

Reply:
True, that would be an alternative. However, its outputs differ from `setuptools_scm` in case of a dirty working copy (not sure if it would matter for CI runs in a clean repo):

![image](https://github.com/appliedAI-Initiative/lakefs-spec/assets/322624/77d83f03-8942-4fca-9ceb-e497e72e543b)